TestApi/TestApi.DAL/Queries/GetUserByIdQuery.cs

```diff
@@ -27,9 +28,16 @@ public GetUserByIdQueryHandler(TestApiDbContext context)
 
         public async Task<User> Handle(GetUserByIdQuery query)
         {
-            return await _context.Users
+            var user = await _context.Users
                 .AsNoTracking()
                 .SingleOrDefaultAsync(x => x.Id == query.Id);
+
+            if (user == null)
```

Comment:
you don't need to throw an exception here. a null would be correct since you're nothing doing anything more with the variable other than returning it in this method

Reply:
Cool, removing that and adjusting the test